deepchem/models/torch_models/layers.py

```diff
@@ -14,36 +15,175 @@ class ScaleNorm(nn.Module):
   >>> input_tensor = torch.tensor([[1.269, 39.36], [0.00918, -9.12]])
-  >>> output_tensor = layer.forward(input_tensor)
+  >>> output_tensor = layer(input_tensor)
   """
 
-  def __init__(self, scale, eps=1e-5):
+  def __init__(self, scale: int, eps: float = 1e-5):
     """Initialize a ScaleNorm layer.
 
     Parameters
     ----------
     scale: Real number or single element tensor
       Scale magnitude.
     eps: float
-      Epsilon value.
+      Epsilon value. Default = 1e-5.
     """
-
     super(ScaleNorm, self).__init__()
     self.scale = nn.Parameter(torch.tensor(math.sqrt(scale)))
     self.eps = eps
 
-  def forward(self, x):
+  def forward(self, x: torch.Tensor):
     norm = self.scale / torch.norm(x, dim=-1, keepdim=True).clamp(min=self.eps)
     return x * norm
+
+
+class MATEmbedding(nn.Module):
+  """Embedding layer to create embedding for inputs.
+
+  In an embedding layer, input is taken and converted to a vector representation for each input.
+  In the MATEmbedding layer, an input tensor is processed through a dropout-adjusted linear layer and the resultant vector is returned.
+
+  References
+  ----------
+  .. [1] Lukasz Maziarka et al. "Molecule Attention Transformer" Graph Representation Learning workshop and Machine Learning and the Physical Sciences workshop at NeurIPS 2019. 2020. https://arxiv.org/abs/2002.08264
+
+  Examples
+  --------
+  >>> import deepchem as dc
+  >>> layer = MATEmbedding(d_input = 1024, d_output = 1024, dropout_p = 0.2)
+  """
+
+  def __init__(self, d_input: int, d_output: int, dropout_p: float):
+    """Initialize a MATEmbedding layer.
+
+    Parameters
+    ----------
+    d_input: int
+      Size of input layer.
+    d_output: int
+      Size of output layer.
+    dropout_p: float
+      Dropout probability for layer.
+    """
+    super(MATEmbedding, self).__init__()
+    self.linear_unit = nn.Linear(d_input, d_output)
+    self.dropout = nn.Dropout(dropout_p)
+
+  def forward(self, x: torch.Tensor):
+    """Computation for the MATEmbedding layer.
+
+    Parameters
+    ----------
+    x: torch.Tensor
+      Input tensor to be converted into a vector.
+    """
+    return self.dropout(self.linear_unit(x))
+
+
+class MATGenerator(nn.Module):
+  """MATGenerator defines the linear and softmax generator step for the Molecular Attention Transformer [1]_.
+
+  In the MATGenerator, a Generator is defined which performs the Linear + Softmax generation step.
+  Depending on the type of aggregation selected, the attention output layer performs different operations.
+
+  References
+  ----------
+  .. [1] Lukasz Maziarka et al. "Molecule Attention Transformer" Graph Representation Learning workshop and Machine Learning and the Physical Sciences workshop at NeurIPS 2019. 2020. https://arxiv.org/abs/2002.08264
+
+  Examples
+  --------
+  >>> import deepchem as dc
+  >>> layer = MATGenerator(hsize = 1024, aggregation_type = 'mean', d_output = 1, n_layers = 1, attn_hidden = 128, attn_out = 4)
```

Comment:
Can you add an example showing how to invoke the layer on input?

Reply:
Added!